### DISCUSSION TITLE
Rendering ansi escape codes and output redirection

### DISCUSSION THREAD

**Discussion by JacekArdanowski** [spectreconsole/spectre.console#997] [Ideas]
Hi,
I was just wondering if it's possible not to print ANSI escape characters when console output is being redirected to a file.
I'm thinking about special option which will prevent writing ANSI with:
`app.exe > output.txt `

In case of ordinary markup text: it is fine, because we can check the property `Console.IsOutputRedirected` and decide whether to use standard Console.Out, AnsiConsole.Markup() or even strip the markup from string.
Although with **AnsiConsole.Write(_IRenderable_)** and output redirection, it is impossible to control this behaviour and output.txt will always have ANSI characters.

So my question is: is it possible to add a flag to AnsiConsole which won't produce any ANSI characters when output from console is being redirected? Thanks.

**Comment by patriksvensson** [comment]
@JacekArdanowski You can set `AnsiConsole.Profile.Capabilities.Ansi` to `false` to prevent ANSI/VT codes from being outputted.

**Reply by JacekArdanowski** [reply]
Thank you! I'm going to see if it helps.

**Reply by jhgbrt** [reply]
This seems to not work for tables, is there another way for that? 

Test code (file based app):

```
#:package Spectre.Console@0.54.0
using Spectre.Console;

AnsiConsole.Profile.Capabilities.Ansi = false;

AnsiConsole.MarkupLine("[bold red]Hello[/], [green]World![/]");
AnsiConsole.MarkupLine("[underline blue]This is underlined blue text.[/]");

var table = new Table();
table.AddColumn("Name");
table.AddColumn("Age");
table.AddRow("Alice", "30");
table.AddRow("Bob", "25");
AnsiConsole.Write(table);
```

When running this in powershell with `dotnet run .\test.cs`, you get full Ansi output (which is ok), when redirecting to a file, the table borders are not replaced by ASCII equivalent characters: 

ΓöîΓöÇΓöÇΓöÇΓöÇΓöÇΓöÇΓöÇΓö¼ΓöÇΓöÇΓöÇΓöÇΓöÇΓöÉ
Γöé Name  Γöé Age Γöé
Γö£ΓöÇΓöÇΓöÇΓöÇΓöÇΓöÇΓöÇΓö╝ΓöÇΓöÇΓöÇΓöÇΓöÇΓöñ
Γöé Alice Γöé 30  Γöé
Γöé Bob   Γöé 25  Γöé
ΓööΓöÇΓöÇΓöÇΓöÇΓöÇΓöÇΓöÇΓö┤ΓöÇΓöÇΓöÇΓöÇΓöÇΓöÿ

**Reply by patriksvensson** [reply]
Seems like an encoding issue. Try changing the file encoding to whatever your terminal's encoding is.
